### PR TITLE
Bomber John map cleanup

### PR DESCRIPTION
**Pull request by abcdefg30**
First step towards reworking this minigame.

This PR additionally adds support for disabling adjacency on the `Building` trait.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
> This PR additionally adds support for disabling adjacency on the Building trait.

That sounds like it should be in a separate PR.

File: OpenRA.Mods.Common/Traits/Buildings/Building.cs [hunk 1]
```diff
@@ -36,7 +36,8 @@ public class BuildingInfo : ITraitInfo, IOccupySpaceInfo, IPlaceBuildingDecorati
 		[Desc("Where you are allowed to place the building (Water, Clear, ...)")]
 		public readonly HashSet<string> TerrainTypes = new HashSet<string>();
 
-		[Desc("The range to the next building it can be constructed. Set it higher for walls.")]
+		[Desc("The range to the next building it can be constructed. Set it higher for walls.",
+			"Set to zero to disable adjacency.")]
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Buildings/Building.cs]: Maybe -1 instead? 0 could also mean to not give adjacency. What you mean is disable adjacency checks though.

**Review by Mailaender — DISMISSED** [review]
:+1: on the map cleanup.

**Comment by abcdefg30** [comment]
Rebased & updated.

**Review by Mailaender — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0bcacd6de6c95295bba5f30d24db9cc062781c32...358d86595d7cd714304dae8ec211b2ce0c1ecd31)

**Pull request merged**